wallet/src/lib.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod wallet;
```

Comment:
Perhaps it makes sense to reexport the `Wallet` struct here, because otherwise using it in other crate would be somewhat awkward: `use wallet::wallet::Wallet`. Maybe `WalletError` can be exported too.

Reply:
while `use wallet::wallet::Wallet` is ugly, I hate reexporting... it's heavily error prone because people tend to forget to reexport when adding new symbols. I can't count how many times I did this mistake. I like the idea of once I export, everything inside is public, and we're done.

Reply:
I'm not sure I understand the reasoning. Sure, it is possible to forget to reexport a new symbol, but it is also possible to forget to make it public. This can easily be fixed by the next person who needs to use this new symbol. I believe that reexports lead to a better and more clear interface: it is easier to see what is available outside of the crate and it prevents exposing everything.

By the way, the current situation with imports is somewhat inconsistent. For example, we have `common::chain::Block` and `common:chain::block::BlockHeader`.

Saying all of that, I still think that it isn't super important, so I'm OK with leaving things as they are.

Reply:
The reasoning is simple: One export is easier than two.

I faced this issue with crates that are in different repos and it was a pain. So I just hate that habit.